pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -286,64 +339,49 @@ func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTempla
 	return &notifier.TestTemplatesResults{}, nil
 }
 
+// StopAndWait is called when the grafana server is instructed to shut down or an org is deleted.
+// In the context of a "remote Alertmanager" is a good heuristic for Grafana is about to shut down or we no longer need you.
```

Comment:
For Grafana, a remote Alertmanager and internal Alertmanager mean two different things even if they share the same interface.

As an example, the method `SaveAndApplyDefaultConfig` doesn't make sense in the context of a Remote Alertmanager as Grafana is not in control of the "recovery" mechanism the original method was designed for.

Hence, the quote of "remote Alertmanager" as in the context of the interface this means way more than "just another alertmanager that's external to grafana".

I have fixed the use of `it's` instead of `is` but I think the comment as is is sound.

Reply:
Great, that's what I thought 👍